python/dgl/distributed/graph_partition_book.py

```diff
@@ -1363,26 +1398,32 @@ class HeteroDataName(object):
 
     def __init__(self, is_node, entity_type, data_name):
         self._policy = NODE_PART_POLICY if is_node else EDGE_PART_POLICY
+        if not is_node:
+            assert isinstance(entity_type, tuple) and len(entity_type) == 3, \
+                "Only canonical etype in tuple is supproted: (str, str, str)."
         self._entity_type = entity_type
         self.data_name = data_name
 
     @property
     def policy_str(self):
         """concatenate policy and entity type into string"""
-        return self._policy + ":" + str(self.get_type())
+        entity_type = self.get_type()
+        if self.is_edge():
+            entity_type = _etype_tuple_to_str(entity_type)
+        return self._policy + POLICY_DELIMITER + entity_type
```

Comment:
Why not directly return `self._policy_str` since you've saved it at L1225.

Reply:
because we need to append the data_name as well